Assets/Scripts/ECS/Movement/Pathfinding/PathfindingSystem.cs

```diff
@@ -23,13 +25,20 @@ protected override void OnCreate()
             base.OnCreate();
             m_EndSimulationEcbSystem = World
                 .GetOrCreateSystem<EndSimulationEntityCommandBufferSystem>();
+
         }
 
         protected override void OnUpdate()
         {
             var commandBuffer = m_EndSimulationEcbSystem.CreateCommandBuffer().ToConcurrent();
-
-            Entities.ForEach((Entity entity, int entityInQueryIndex,
+            NativeArray<bool> grid = GameZone.walkableTiles;
+            var gridSizeArray = new NativeArray<int2>(1, Allocator.TempJob);
+            gridSizeArray[0] = new int2(GameZone.tiles.GetLength(0), GameZone.tiles.GetLength(1));
```

Comment:
Fungerar det inte att bara skapa en vanlig int2-variabel istället för NativeArray här?

Reply:
Nä den klagade på det

Reply:
Hmm, konstigt. Kunde du tyda någonting från error-meddelandet?

Reply:
"Entities.WithReadOnly is called with an argument gridSize of unsupported type Unity.Mathematics.int2. It can only be called with an argument that is marked with [NativeContainerAttribute]."

Men i [dokumentationen här](https://docs.unity3d.com/Packages/com.unity.entities@0.8/manual/ecs_entities_foreach.html) står det:
"Only native containers and blittable types can be captured."

Vad jag förstår är int2 blittable, så det borde gå, men error-meddelandet verkar ju peka på att det krävs en NativeContainer.

Reply:
Det blir samma error om man skickar in en vanlig int också, och den är ju definitivt blittable.

Reply:
Problemet är nog att WithReadOnly bara används när det gäller native containers. Blittable types borde fungera utan det anropet.

Reply:
Det verkar göra det